openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -640,13 +640,18 @@ private Map toTupleSubMap(Object[] a, String entryKey, String entryValue) {
 
             boolean isPublisher = false;
             boolean isObservable = false;
+            boolean isNullable = false;
 
             // StreamingFileUpload implements Publisher, but it should be not considered as a Publisher in the spec file
             if (!type.isAssignable("io.micronaut.http.multipart.StreamingFileUpload") && isContainerType(type)) {
                 isPublisher = type.isAssignable(Publisher.class.getName()) && !type.isAssignable("reactor.core.publisher.Mono");
                 isObservable = type.isAssignable("io.reactivex.Observable") && !type.isAssignable("reactor.core.publisher.Mono");
                 type = type.getFirstTypeArgument().orElse(null);
+            } else if (type.isAssignable("java.util.Optional") || type.isAssignable("com.google.common.base")) {
+               isNullable = true;
+               type = type.getFirstTypeArgument().orElse(null);
```

Comment:
This seems it can be only once after the if/else block

Reply:
sorry do you mean that we should recursively resolve it? so `Optional<Optional<X>> ` resolves to `nullable: true` and type `X` ?

Reply:
No, I just mean the line `type = type.getFirstTypeArgument().orElse(null);`, which is duplicated at the end of both branches of the if/else, could be left after the if/else block.

Reply:
it's an `else if` so it wouldn't be correct to move it outside the block.

Reply:
Ugh, you are right, ignore me